indra/newview/llscreenchannel.cpp

```diff
@@ -301,24 +301,28 @@ void LLScreenChannel::addToast(const LLToast::Params& p)
 
 	toast->setOnFadeCallback(boost::bind(&LLScreenChannel::onToastFade, this, _1));
 	toast->setOnToastDestroyedCallback(boost::bind(&LLScreenChannel::onToastDestroyed, this, _1));
-	if(mControlHovering)
+	if (mControlHovering)
 	{
 		toast->setOnToastHoverCallback(boost::bind(&LLScreenChannel::onToastHover, this, _1, _2));
 		toast->setMouseEnterCallback(boost::bind(&LLScreenChannel::stopToastTimer, this, toast));
 		toast->setMouseLeaveCallback(boost::bind(&LLScreenChannel::startToastTimer, this, toast));
 	}
-	
-	if(show_toast)
+
+	if (show_toast)
 	{
 		mToastList.push_back(new_toast_elem);
-		if(p.can_be_stored)
+		if (toast->isModal())
+		{
+			toast->openFloater(LLSD());
```

Comment:
Notifications can be unique so at minimum this should require a ke, not LLSD(). This is also likely to break 'ignore' mechanics since notifications will be opened regardles of 'ignore' being set.

Reply:
I don't see why modal toats should be handled separately and need an openFloater, I suggest overloading modal toasts' setVisible instead to call hideMenus